### DISCUSSION TITLE
Nightscout on Google Cloud - 502 Bad Gateway

### DISCUSSION THREAD

**Discussion by Ruhrtyp1** [NightscoutFoundation/xDrip#3169] [Bug reports]
Hello Folks,

after updating the recent Platform, I get the '502 - bad gateway' error. I wasn't able to resolve it neither by restarting the platform nor bootstrap or running the installation phase 2...

The status seems to be ok, as you can see in the screenshot.

Any ideas to get me out of this misery - I have a doctors appointment on Wednesday - he's eager to see my Results regarding the last three months...

Thank you in advance!
<img width="900" alt="screenshot_status_nightscout" src="https://github.com/NightscoutFoundation/xDrip/assets/149953521/5508cc82-d968-4711-904d-f29828090cbf">

**Comment by Navid200** [comment]
Hi,

If you log into your FreeDNS account and go to the subdomains page, how many subdomains do you have?
Do they all look bold or does the one with trouble look different?

**Reply by Ruhrtyp1** [reply]
Hi,

the One I am using looks - how you say it - normal. There is the domain and there is the Destination. Nothing appears unusual...

**Comment by Navid200** [comment]
Do you mind creating a second subdomain?
It could be on a different domain.  

Please have a look at the following screenshot.
![Screenshot 2023-10-26 103807](https://github.com/NightscoutFoundation/xDrip/assets/51497406/cd078271-e42b-4cfa-b17f-1c17702bc5f4)

Do you see the three at the bottom?  
There are times something is wrong with a subdomain and FreeDNS only shows it this way.  This is not very obvious and I am hoping to find a way to improve this.

If you create a new subdomain and run phase 2 with it, and you can press escape when it asks you for changing API_SECRET to keep it unchanged, everything should be sorted out.

**Reply by Navid200** [reply]
The three subdomains at the bottom are not functional.  And the only way to tell is that they are not shown in boldface.

**Reply by Ruhrtyp1** [reply]
All right. I will try my best and respond.

**Reply by Navid200** [reply]
Was everything working fine?
Then, you decided to update platform and you did.  And it stopped working?

**Reply by Navid200** [reply]
If everything was working fine and the only thing you did was to update platform, I suggest running phase 1 again.

But, if things weren't working for a while and you decided to update platform hoping it may fix things, then, it seems something may be wrong with the subdomain.

**Reply by Ruhrtyp1** [reply]
So, I did change the subdomain to another top-domain, and the 502 - bad gateway is still there...

Yes, before the Update everything was working - last time I signed in via my domain was yesterday.

**Reply by Navid200** [reply]
Can you run phase 1?

**Comment by Ruhrtyp1** [comment]
I'm back in business - hooray! Installing Phase 1 did the job!

Thanks for your Help, Navid! Have a great Sunday!

**Reply by Navid200** [reply]
You are welcome.

Just in case you want to switch back to the original hostname, you can.
https://navid200.github.io/xDrip/docs/Nightscout/ChangeHostname.html

And if you are happy with the new hostname, that's fine too.
If you want to switch back and run into trouble, please let me know.
